source/Objects/Weapons/Weapon2D.gd

```diff
@@ -21,18 +20,19 @@ func _shoot() -> void:
 	if not _cooldown_timer.is_stopped():
 		return
 	_cooldown_timer.start()
-	
+
 	look_at(target.global_position)
 	_animation_player.play("shoot")
-	
+
 	_bullet_spawner.spawn_scene = bullet_scene
-	var bullet: Bullet = _bullet_spawner.spawn()
+	var bullet: Bullet = yield(_bullet_spawner.spawn(), "completed")
```

Comment:
That's really nice. I tried something similar but I couldn't get the actual `return` of the method, instead just the function state.

Thanks a lot!

Reply:
Yep; all function states emit the completed signal. It'll be even nicer in 4.0 when we can just do `await _bullet_spawner.spawn()`